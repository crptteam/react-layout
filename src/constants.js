export const COLORS = {
  dark: 'dark',
  light: 'light',
  default: 'light'
}