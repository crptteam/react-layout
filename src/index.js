import Column from './components/Column';
import Container from './components/Container';
import Content from './components/Content';
import Footer from './components/Footer';
import Header from './components/Header';
import Layout from './components/Layout';
import InlineWrap from './styled/InlineWrap';
import {COLORS} from './constants';

export { Column, Container, Content, Header, Footer, Layout, InlineWrap, COLORS };

export default Layout;