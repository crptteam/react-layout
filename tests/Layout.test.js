import React from "react";
import { Layout, Header, Footer, Container, Content, Column } from "../src/";
import { mount } from "enzyme";

describe("Input", () => {
  it("Should renders without problems", () => {
    const wrapper = mount(
      <Layout>
        <Header>I am header.</Header>
        <Content>
          <Container>
            <Column>Column one.</Column>
            <Column>Column two.</Column>
            <Column>Column three.</Column>
          </Container>
        </Content>
        <Footer>I am footer.</Footer>
      </Layout>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
