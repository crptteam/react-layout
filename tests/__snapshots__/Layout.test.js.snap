// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should renders without problems 1`] = `
<WithTheme(Layout)>
  <Layout>
    <OuterLayout>
      <div
        className="OuterLayout-grMLHR csRMKR"
      >
        <WithTheme(Header)>
          <Header>
            <OuterHeader>
              <div
                className="OuterHeader-dPqjSd iRMKMX"
              >
                <InnerHeader>
                  <InnerHeader__Elem
                    background="#212C42"
                    color="#FFFFFF"
                  >
                    <div
                      className="InnerHeader__Elem-cHbucc kFLLHQ"
                      color="#FFFFFF"
                    >
                      I am header.
                    </div>
                  </InnerHeader__Elem>
                </InnerHeader>
              </div>
            </OuterHeader>
          </Header>
        </WithTheme(Header)>
        <WithTheme(Content)>
          <Content>
            <OuterContent>
              <div
                className="OuterContent-koFsIb dCBBnm"
              >
                <InnerContent>
                  <InnerContent__Elem
                    background="#E5E5E5"
                    color="#212C42"
                  >
                    <div
                      className="InnerContent__Elem-eYwcsk ekVinH"
                      color="#212C42"
                    >
                      <WithTheme(Container)>
                        <Container>
                          <OuterContainer>
                            <OuterContainer__Elem>
                              <div
                                className="OuterContainer__Elem-iYocSL jmEptY"
                              >
                                <WithTheme(Column)
                                  first={true}
                                  key=".0"
                                  last={false}
                                >
                                  <Column
                                    first={true}
                                    last={false}
                                  >
                                    <OuterColumn
                                      first={true}
                                      last={false}
                                    >
                                      <OuterColumn__Elem
                                        first={true}
                                        last={false}
                                      >
                                        <div
                                          className="OuterColumn__Elem-bwpHlP hkrUAK"
                                        >
                                          <InnerColumn__Inner>
                                            <div
                                              className="InnerColumn__Inner-fGuEuo kQnsUP"
                                            >
                                              Column one.
                                            </div>
                                          </InnerColumn__Inner>
                                        </div>
                                      </OuterColumn__Elem>
                                    </OuterColumn>
                                  </Column>
                                </WithTheme(Column)>
                                <WithTheme(Column)
                                  first={false}
                                  key=".1"
                                  last={false}
                                >
                                  <Column
                                    first={false}
                                    last={false}
                                  >
                                    <OuterColumn
                                      first={false}
                                      last={false}
                                    >
                                      <OuterColumn__Elem
                                        first={false}
                                        last={false}
                                      >
                                        <div
                                          className="OuterColumn__Elem-bwpHlP hkrUAK"
                                        >
                                          <InnerColumn__Inner>
                                            <div
                                              className="InnerColumn__Inner-fGuEuo kQnsUP"
                                            >
                                              Column two.
                                            </div>
                                          </InnerColumn__Inner>
                                        </div>
                                      </OuterColumn__Elem>
                                    </OuterColumn>
                                  </Column>
                                </WithTheme(Column)>
                                <WithTheme(Column)
                                  first={false}
                                  key=".2"
                                  last={true}
                                >
                                  <Column
                                    first={false}
                                    last={true}
                                  >
                                    <OuterColumn
                                      first={false}
                                      last={true}
                                    >
                                      <OuterColumn__Elem
                                        first={false}
                                        last={true}
                                      >
                                        <div
                                          className="OuterColumn__Elem-bwpHlP hkrUAK"
                                        >
                                          <InnerColumn__Inner>
                                            <div
                                              className="InnerColumn__Inner-fGuEuo kQnsUP"
                                            >
                                              Column three.
                                            </div>
                                          </InnerColumn__Inner>
                                        </div>
                                      </OuterColumn__Elem>
                                    </OuterColumn>
                                  </Column>
                                </WithTheme(Column)>
                              </div>
                            </OuterContainer__Elem>
                          </OuterContainer>
                        </Container>
                      </WithTheme(Container)>
                    </div>
                  </InnerContent__Elem>
                </InnerContent>
              </div>
            </OuterContent>
          </Content>
        </WithTheme(Content)>
        <WithTheme(Footer)>
          <Footer>
            <OuterFooter>
              <div
                className="OuterFooter-zNYqb gaDniJ"
              >
                <InnerFooter>
                  <InnerFooter__Elem
                    background="#212C42"
                    color="#FFFFFF"
                  >
                    <div
                      className="InnerFooter__Elem-cVCqvV cKTgUr"
                      color="#FFFFFF"
                    >
                      I am footer.
                    </div>
                  </InnerFooter__Elem>
                </InnerFooter>
              </div>
            </OuterFooter>
          </Footer>
        </WithTheme(Footer)>
      </div>
    </OuterLayout>
  </Layout>
</WithTheme(Layout)>
`;
